lib/EnsEMBL/REST/Model/Variation.pm

```diff
@@ -465,20 +465,33 @@ sub fetch_population_infos {
 }
 
 sub fetch_consequence_types {
-  my ($self) = @_;
+  my ($self, $rank) = @_;
 
   my @consequence_types = ();
 
   my $oc;
   foreach my $key(keys %OVERLAP_CONSEQUENCES) {
     $oc = $OVERLAP_CONSEQUENCES{$key};
-    push @consequence_types,
+
+    if (defined $rank && $rank == 1) {
+        push @consequence_types,
+          {
+            SO_term => $oc->SO_term,
+            SO_accession => $oc->SO_accession,
+            label => $oc->label,
+            description =>  $oc->description,
+            consequence_ranking => $oc->rank
+          }
+    }
+    else {
+      push @consequence_types,
         {
           SO_term => $oc->SO_term,
           SO_accession => $oc->SO_accession,
           label => $oc->label,
           description =>  $oc->description
         }
+    }
```

Comment:
The duplication should be avoided by setting the default stuff, and then adding the specifics... something like:
```
my $item = {
  SO_term      => $oc->SO_term,
  SO_accession => $oc->SO_accession,
  label        => $oc->label,
  description  => $oc->description
};

if ($rank) {
  $item->{'consequence_ranking'} = $oc->rank;
}

push @consequence_types, $item;
```

Reply:
Thanks for the suggestions.